examples/sparse/hetero-rgcn.py

```diff
@@ -0,0 +1,498 @@
+"""Modeling Relational Data with Graph Convolutional Networks
+Paper: https://arxiv.org/abs/1703.06103
+Reference Code: https://github.com/tkipf/relational-gcn
+"""
+import argparse
+import time
+
+import numpy as np
+import torch as th
+import torch.nn as nn
+import torch.nn.functional as F
+
+from dgl.data.rdf import AIFBDataset, AMDataset, BGSDataset, MUTAGDataset
+
+
+from collections import defaultdict
+
+import dgl
+import dgl.function as fn
+import dgl.nn as dglnn
+import dgl.sparse as dglsp
+
+import torch as th
+import torch.nn as nn
+import torch.nn.functional as F
+import tqdm
+
+
+class RelGraphEmbed(nn.Module):
+    r"""Embedding layer for featureless heterograph."""
+
+    def __init__(
+        self, g, embed_size, embed_name="embed", activation=None, dropout=0.0
+    ):
+        super(RelGraphEmbed, self).__init__()
+        self.g = g
+        self.embed_size = embed_size
+        self.embed_name = embed_name
+        self.activation = activation
+        self.dropout = nn.Dropout(dropout)
+
+        # create weight embeddings for each node for each relation
+        self.embeds = nn.ParameterDict()
+        for ntype in g.ntypes:
+            embed = nn.Parameter(th.Tensor(g.num_nodes(ntype), self.embed_size))
+            nn.init.xavier_uniform_(embed, gain=nn.init.calculate_gain("relu"))
+            self.embeds[ntype] = embed
+
+    def forward(self, block=None):
+        """Forward computation
+
+        Parameters
+        ----------
+        block : DGLGraph, optional
+            If not specified, directly return the full graph with embeddings stored in
+            :attr:`embed_name`. Otherwise, extract and store the embeddings to the block
+            graph and return.
+
+        Returns
+        -------
+        DGLGraph
+            The block graph fed with embeddings.
+        """
+        return self.embeds
+
+
+class HeteRGConvLayer(nn.Module):
+    r"""Relational graph convolution layer.
+
+    Parameters
+    ----------
+    in_feat : int
+        Input feature size.
+    out_feat : int
+        Output feature size.
+    rel_names : list[str]
+        Relation names.
+    num_bases : int, optional
+        Number of bases. If is none, use number of relations. Default: None.
+    weight : bool, optional
+        True if a linear layer is applied after message passing. Default: True
+    bias : bool, optional
+        True if bias is added. Default: True
+    activation : callable, optional
+        Activation function. Default: None
+    self_loop : bool, optional
+        True to include self loop message. Default: False
+    dropout : float, optional
+        Dropout rate. Default: 0.0
+    """
+
+    def __init__(
+        self,
+        in_feat,
```

Comment:
To simplify this example, only keep the `in_feat`, `out_feat`, `rel_names` here.